File: Problems/Easy/PrintInOrder/yash_printInOrder.py
class Foo:
    def __init__(self):
        self.tmp = 0 
        # Kinda using a variable to lock and release.


    def first(self, printFirst: 'Callable[[], None]') -> None:
        while(self.tmp != 0):
            continue
            
        # printFirst() outputs "first". Do not change or remove this line.
        printFirst()
        self.tmp = self.tmp | 1


    def second(self, printSecond: 'Callable[[], None]') -> None:
        while (self.tmp != 1):
            continue
        # printSecond() outputs "second". Do not change or remove this line.
        printSecond()
        self.tmp = self.tmp | 11


    def third(self, printThird: 'Callable[[], None]') -> None:
        while (self.tmp != 11):
            continue
        # printThird() outputs "third". Do not change or remove this line.
        printThird()
        self.tmp = self.tmp | 111

File: Problems/Easy/PrintInOrder/yash_printInOrder.c
typedef struct {
    // User defined data may be declared here.
    unsigned char tmp;
} Foo;

Foo* fooCreate() {
    Foo* obj = (Foo*) malloc(sizeof(Foo));
    obj -> tmp = 0;
    // Initialize user defined data here.
    
    return obj;
}

void first(Foo* obj) {
    while (obj -> tmp != 0)
        continue;
    // printFirst() outputs "first". Do not change or remove this line.
    printFirst();
    obj -> tmp |= 0b1;
}

void second(Foo* obj) {
    while (obj -> tmp != 1)
        continue;
    // printSecond() outputs "second". Do not change or remove this line.
    printSecond();
    obj -> tmp |= 0b11;
}

void third(Foo* obj) {
    while (obj -> tmp != 0b11)
        continue;
    // printThird() outputs "third". Do not change or remove this line.
    printThird();
    
}

void fooFree(Foo* obj) {
    // User defined data may be cleaned up here.
    free(obj);
}